torchgeo/datamodules/inria.py

```diff
@@ -0,0 +1,236 @@
+        self,
+        root_dir: str,
+        batch_size: int = 32,
+        num_workers: int = 0,
+        val_split_pct: float = 0.1,
+        test_split_pct: float = 0.1,
+        patch_size: Union[int, Tuple[int, int]] = 512,
+        num_patches_per_tile: int = 32,
+        augmentations: K.AugmentationSequential = DEFAULT_AUGS,
+        predict_on: str = "test",
+    ) -> None:
+        """Initialize a LightningDataModule for InriaAerialImageLabeling based DataLoaders.
+
+        Args:
+            root_dir: The ``root`` arugment to pass to the InriaAerialImageLabeling
+                Dataset classes
+            batch_size: The batch size used in the train DataLoader
+                (val_batch_size == test_batch_size == 1)
+            num_workers: The number of workers to use in all created DataLoaders
+            val_split_pct: What percentage of the dataset to use as a validation set
+            test_split_pct: What percentage of the dataset to use as a test set
+            patch_size: Size of random patch from image and mask (height, width)
+            num_patches_per_tile: Number of random patches per sample
+            augmentations: Default augmentations applied
+            predict_on: Directory/Dataset of images to run inference on
+        """
+        super().__init__()  # type: ignore[no-untyped-call]
+        self.root_dir = root_dir
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.val_split_pct = val_split_pct
+        self.test_split_pct = test_split_pct
+        self.patch_size = cast(Tuple[int, int], _pair(patch_size))
+        self.num_patches_per_tile = num_patches_per_tile
+        self.augmentations = augmentations
+        self.predict_on = predict_on
+        self.random_crop = K.AugmentationSequential(
+            K.RandomCrop(self.patch_size, p=1.0, keepdim=False),
+            data_keys=["input", "mask"],
+        )
+
+    def patch_sample(self, sample: Dict[str, Any]) -> Dict[str, Any]:
+        """Extract patches from single sample."""
+        assert sample["image"].ndim == 3
+        _, h, w = sample["image"].shape
+
+        padding = compute_padding((h, w), self.patch_size)
+        sample["original_shape"] = (h, w)
+        sample["patch_shape"] = self.patch_size
+        sample["padding"] = padding
+        sample["image"] = extract_tensor_patches(
+            sample["image"].unsqueeze(0),
+            self.patch_size,
+            self.patch_size,
+            padding=padding,
+        )
+        sample["image"] = rearrange(sample["image"], "() t c h w -> t () c h w")
+        return sample
+
+    def preprocess(self, sample: Dict[str, Any]) -> Dict[str, Any]:
+        """Transform a single sample from the Dataset."""
+        # RGB is int32 so divide by 255
+        sample["image"] = sample["image"] / 255.0
+        sample["image"] = torch.clip(sample["image"], min=0.0, max=1.0)
+
+        # This is pointless since it will get squeezed out anyway
```

Comment:
I assume there's a reason we do it despite it being squeezed out later? Might want to elaborate on that